### PR TITLE
[cherry-pick]Failing the velero backup if pv is not supported

### PR DESCRIPTION
**Pull request by liuy1vmware**
Before this change, velero backup just skips the pv if the pv type is not supported
Afte this change, when the migration feature gate is enabled, an unsupported pv type fails the whole velero backup

Signed-off-by: Yang Liu <liuy1@liuy1-a02.vmware.com>

Co-authored-by: Yang Liu <liuy1@liuy1-a02.vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
This crossport includes following 2 changes
Cherry-pick #460: Return error to make velero backup partially failed, if the pv is not a supported volume type
Cherry-pick #461: When storageClass is empty string, treat it as no storageClass is given
```

### CODE REVIEW

**Comment by xing-yang** [comment]
Can you remove PR number from the commit messages?

**Comment by xing-yang** [comment]
Can you add "Cherry-pick #<original PR number>" in release notes?

**Comment by liuy1vmware** [comment]
> 

done

File: pkg/plugin/backup_pvc_action_plugin.go [hunk 1]
```diff
@@ -100,15 +100,27 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
 		return nil, nil, errors.WithStack(err)
 	}
 
+	// Do nothing if restic is used to backup this PV
```
Comment (xing-yang) [inline pkg/plugin/backup_pvc_action_plugin.go:103]: Can you remove #460 from this commit message?

File: pkg/backupdriver/backup_driver_controller.go [hunk 2]
```diff
@@ -289,7 +289,7 @@ func (ctrl *backupDriverController) cloneFromSnapshot(cloneFromSnapshot *backupd
 		}
 		return err
 	}
-	if pvc.Spec.StorageClassName != nil {
+	if pvc.Spec.StorageClassName != nil && (*pvc.Spec.StorageClassName) != ""{
```
Comment (xing-yang) [inline pkg/backupdriver/backup_driver_controller.go:292]: Can you remove #461 from the commit message?

File: pkg/plugin/backup_pvc_action_plugin.go (hunk 1 above)
Comment (liuy1vmware) [reply pkg/plugin/backup_pvc_action_plugin.go:103]: done

**Review by xing-yang — APPROVED** [review]
LGTM

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/723/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/724/

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/725/

**Pull request merged**